### PR TITLE
std.path: make .ptr access trusted

### PR DESCRIPTION
**Pull request by WalterBright**
blocker for https://github.com/dlang/dmd/pull/5860

### CODE REVIEW

File: std/path.d [hunk 1]
```diff
@@ -3269,7 +3269,7 @@ body
                             /* Match for:
                              *   pattern[pi0..pi-1] ~ pattern[piRemain..$]
                              */
-                            if (pattmp.ptr == null)
+                            if (() @trusted { return pattmp.ptr; }() == null)
```
Comment (andralex) [inline std/path.d]: This is easy to write safely as `pattmp is null`. What am I missing?

Comment (WalterBright) [inline std/path.d]: array.ptr is simply rejected by the safety check in the compiler.

Comment (andralex) [inline std/path.d]: There is no `.ptr` in `pattmp is null`

Comment (WalterBright) [inline std/path.d]: The trouble being that `==null` can appear in arbitrarily complex forms.

Comment (andralex) [inline std/path.d]: There is no `==` in `pattmp is null`

Comment (WalterBright) [inline std/path.d]: oh I see what you mean. The compiler generates some rather stupid code for `array is null`. Should be a topic for another issue.

Comment (andralex) [inline std/path.d]: Let's use this as a forcing function for that. `e1 is e2` should be `@safe ` for any two slice expressions `e1` and `e2`.

Comment (WalterBright) [inline std/path.d]: I remember some argument a while back on what `array == null` should be. Don't remember the details.

Comment (andralex) [inline std/path.d]: If you don't remember the details it doesn't count. `e is null` is `@safe` for any slice expression `e`.

File: std/path.d [hunk 2]
```diff
@@ -3269,7 +3269,7 @@ body
                             /* Match for:
                              *   pattern[pi0..pi-1] ~ pattern[piRemain..$]
                              */
-                            if (pattmp.ptr == null)
+                            if (pattmp == null)
```
Comment (andralex) [inline std/path.d]: `is` is better but I'll choose my battles

**Pull request merged**